src/db/tables/subscriber_email_preferences.ts

```diff
@@ -0,0 +1,246 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import createDbConnection from "../connect.js";
+import { logger } from "../../app/functions/server/logging";
+import { captureException } from "@sentry/node";
+
+const knex = createDbConnection();
+
+// NOTE: The "subscriber_email_preferences" table only has attributes for free reports
+// TODO: modify the CRUD utils after MNTOR-3557
+interface SubscriberFreeEmailPreferencesInput {
+  unsubscribe_token?: string;
+  monthly_monitor_report_free?: boolean;
+  monthly_monitor_report_free_at?: Date;
+}
+
+interface SubscriberPlusEmailPreferencesInput {
+  monthly_monitor_report?: boolean;
+  monthly_monitor_report_at?: Date;
+}
+
+interface SubscriberEmailPreferencesOutput {
+  id?: number;
+  primary_email?: string;
+  unsubscribe_token?: string;
+  monthly_monitor_report_free?: boolean;
+  monthly_monitor_report_free_at?: Date;
+  monthly_monitor_report?: boolean;
+  monthly_monitor_report_at?: Date;
+}
+
+// TODO: modify after MNTOR-3557 - pref currently lives in two tables
+// this function only adds email prefs for free reports
+async function addEmailPreferenceForSubscriber(
+  subscriberId: number,
+  preference: SubscriberFreeEmailPreferencesInput,
+  fieldsToOverwrite: (keyof SubscriberFreeEmailPreferencesInput)[],
+) {
+  logger.info("add_email_preference_for_subscriber", {
+    subscriberId,
+    preference,
+  });
+
+  let res;
+  try {
+    res = await knex("subscriber_email_preferences")
+      .insert({
+        subscriber_id: subscriberId,
+        unsubscribe_token: preference.unsubscribe_token || "",
+        monthly_monitor_report_free:
+          preference.monthly_monitor_report_free ?? true,
+        monthly_monitor_report_free_at:
+          preference.monthly_monitor_report_free_at ?? null,
+      })
+      .onConflict("subscriber_id")
+      .merge(fieldsToOverwrite)
```

Comment:
Oh also, I called this function to be able to test out unsubscription, and it threw an error unless I commented out these two lines. Possibly due to passing an empty array as the third argument, in which case: maybe some JSDoc explaining what it's supposed to be used for could be helpful? It wasn't clear to me 😅

Reply:
added a note